lib/main.dart

```diff
@@ -3,26 +3,28 @@ import 'dart:async';
 import 'package:firebase_core/firebase_core.dart';
 import 'package:flutter/material.dart';
 import 'package:provider/provider.dart';
-import 'package:test_app/Screens/LoginPage.dart';
-import 'package:test_app/Screens/NavBar.dart';
-import 'package:test_app/Screens/splash.dart';
-import 'package:test_app/provider/app.dart';
-import 'package:test_app/provider/product.dart';
-import 'package:test_app/provider/user.dart';
+import 'package:book_rent_app_chapter3/Screens/LoginPage.dart';
+import 'package:book_rent_app_chapter3/Screens/NavBar.dart';
+import 'package:book_rent_app_chapter3/Screens/splash.dart';
+import 'package:book_rent_app_chapter3/provider/app.dart';
+import 'package:book_rent_app_chapter3/provider/product.dart';
+import 'package:book_rent_app_chapter3/provider/user.dart';
 import 'package:flutter/services.dart';
 
 Future<void> main() async {
   final binding = WidgetsFlutterBinding.ensureInitialized();
   // Preload all assets to prevent flash when they are loaded.
   binding.deferFirstFrame();
   binding.addPostFrameCallback((_) {
-    BuildContext context = binding.renderViewElement;
+    BuildContext context = binding.renderViewElement!;
     if (context != null) {
       Splash();
     }
     binding.allowFirstFrame();
   });
-  await Firebase.initializeApp();
+  await Firebase.initializeApp(
+    options: FirebaseOptions(apiKey: "AIzaSyCdQgjJ7kTZ_NxrW-1ReuvInOYyXJTwRTA", appId: "1:154108809680:web:f0f9b39d6f5bdbfbfb29d5", messagingSenderId: "154108809680", projectId: "bookrentapp"),
+  );
```

Comment:
```suggestion
  if (Firebase.apps.isEmpty) {
    await Firebase.initializeApp();
  }else {
    Firebase.app(); // if already initialized, use that one
  }
```

Why are these API keys used? I found an issue with this code and the StackOverflow solution worked :D

Reply:
I am getting this error if I don't pass FirebaseOptions:

```
Error: Assertion failed: file:///C:/Users/Harshit/AppData/Local/Pub/Cache/hosted/pub.dartlang.org/firebase_core_web-1.5.3/lib/src/firebase_core_web.dart:273:11
options != null
"FirebaseOptions cannot be null when creating the default app."
    at Object.throw_ [as throw] (http://localhost:60983/dart_sdk.js:5063:11)
    at Object.assertFailed (http://localhost:60983/dart_sdk.js:4988:15)
    at firebase_core_web.FirebaseCoreWeb.new.initializeApp (http://localhost:60983/packages/firebase_core_web/firebase_core_web.dart.lib.js:252:42)
    at initializeApp.next (<anonymous>)
    at http://localhost:60983/dart_sdk.js:40192:33
    at _RootZone.runUnary (http://localhost:60983/dart_sdk.js:40062:59)
    at _FutureListener.thenAwait.handleValue (http://localhost:60983/dart_sdk.js:34983:29)
    at handleValueCallback (http://localhost:60983/dart_sdk.js:35551:49)
    at Function._propagateToListeners (http://localhost:60983/dart_sdk.js:35589:17)
    at _Future.new.[_completeWithValue] (http://localhost:60983/dart_sdk.js:35437:23)
    at async._AsyncCallbackEntry.new.callback (http://localhost:60983/dart_sdk.js:35458:35)
    at Object._microtaskLoop (http://localhost:60983/dart_sdk.js:40330:13)
    at _startMicrotaskLoop (http://localhost:60983/dart_sdk.js:40336:13)
    at http://localhost:60983/dart_sdk.js:35811:9
```

Reply:
Is this while firing up the app on the web? Or on Android too?

Reply:
@Saransh-cpp Weird, it is working fine on Android, but giving this error if I try to run on chrome without passing FirebaseOptions.

Reply:
@letsintegreat, as I mentioned on Discord, this does apparently require the `options` argument. Please add it back in.

Reply:
> @letsintegreat, as I mentioned on Discord, this does apparently require the `options` argument. Please add it back in.

Done!

Reply:
For some reason, it still gives me the same error. You'll probably need to debug this a bit. Works fine on the web!

Reply:
@Saransh-cpp Yeah, I have fixed it! Now the app's running on both android and web.